### DISCUSSION TITLE
Better separation between single node and grouped nodes

### DISCUSSION THREAD

**Discussion by shadycuz** [netdata/netdata#15623] [Ideas]
Hello, 

I'm trying out the new Parent/Child feature. I find the current dashboard hard to use. It says "System Overview" but if you are on a parent it will show metrics from multiple nodes. It's also in the "local" space, which also tricks my lizard brain into thinking it's showing metrics from a single system and not multiple systems. Even worse, none of the charts make sense to me. For example the CPU will be showing the "average" of the nodes. Not really that helpful since I cant filter the nodes into smaller groups #15620 . So I constantly find my self changing the "group by" to `nodes` and then "line". 

My suggestions:
* Put children in a new Space, possibly named `children`. 
  *  Long term you probably need too add a label to the [API_KEY] section. That way you could create an API key for "prod" and one for "non-prod". Then when you view the parent dashboard, you would have a space for prod and a space for non-prod?
* Allow filtering of nodes by labels. This is needed so that I can label all my machines correctly and then view them as groups that makes sense. 
*  Use a slightly different dashboard for viewing groups of nodes. vs single node. I think all you need to do is change the text from "Sytem Overview" to "${SPACE} Overview". Assuming you add labels to the API_KEY section. That way by default you would get "Children Overview". Or if you used a label like "Prod" for your API_KEY you would get "Prod Overview". 
    * Make the default Group By set to "node" and use the "line" chart.

**Comment by hugovalente-pm** [comment]
hi @shadycuz ,

thank you for the feedback, let me try to comment/answer to it. I'll be using our demo site as example - https://sanfrancisco.my-netdata.io/

> "System Overview" but if you are on a parent it will show metric

the Overview and Single Node view tab dashboards are the same, to this top section is really a systems wide overview. if you have multiple nodes you can easily spot that on some of the summary charts 
![image](https://github.com/netdata/netdata/assets/82235632/f54a1470-9880-4461-917a-2b4ec051cf40)

> It's also in the "local" space, which also tricks my lizard brain into thinking it's showing metrics from a single system and not multiple systems

this is the name of the virtual room that highlights that this about your local agent. the number of nodes it shows will depend if you have children streaming to it but it maps to the parent agent you're accessing in your case

the naming we have:
* Virtual space: hostname, e.g. `sanfrancisconetdatarocks`
* Virtual room, `local`

![image](https://github.com/netdata/netdata/assets/82235632/79abac71-9539-481b-8367-b25a2a9cf6a9)

> For example the CPU will be showing the "average" of the nodes. Not really that helpful since I cant filter the nodes into smaller groups https://github.com/netdata/netdata/discussions/15620

That request about filtering by label is really something we need to bring in, in the meantime we hope the NIDL framework helps you slice and dice the data as you need and provide some summaries.
Example opening this on the Total CPU Utilization chart you can see that on "systems" you get a summary that allows you to see which systems (nodes) contribute more in terms of volume to the data you're seeing - you can combine this with the grouping and filtering options

![image](https://github.com/netdata/netdata/assets/82235632/db80551e-27aa-441a-bd67-8d359bcb650a)

> Put children in a new Space, possibly named children.

This would defeat the purpose to allow a parent to work as an aggregated infrastructure view of all the node it has data for, which was one of the main reasons it lead us to bring this UI to the agent dashboard

> Long term you probably need too add a label to the [API_KEY] section. That way you could create an API key for "prod" and one for "non-prod". Then when you view the parent dashboard, you would have a space for prod and a space for non-prod?

What do you mean? You wanted to have a single parent with child nodes from prod and non-prod?
In this case, it seems it would be more proper to have dedicated parents for each environment, no?

> Allow filtering of nodes by labels. This is needed so that I can label all my machines correctly and then view them as groups that makes sense.

Agreed, this is something we need to work on

> Use a slightly different dashboard for viewing groups of nodes. vs single node. I think all you need to do is change the text from "Sytem Overview" to "${SPACE} Overview". Assuming you add labels to the API_KEY section. That way by default you would get "Children Overview". Or if you used a label like "Prod" for your API_KEY you would get "Prod Overview".
Make the default Group By set to "node" and use the "line" chart.

The spaces point I mentioned above about the default Group by and chart I guess it really depends on what users prefer, for that we have this feature request that we will also try to tackle https://github.com/netdata/netdata-cloud/issues/789

**Comment by shadycuz** [comment]
> the Overview and Single Node view tab dashboards are the same, to this top section is really a systems wide overview. if you have multiple nodes you can easily spot that on some of the summary charts

Yeah I figured it out, but I know I'm not the only one that has complained. My suggestion is to make the dashboards slightly different so thats its easier to distigush a single node dashboard from multiple nodes. It could be as simple as changing "System Overview" to "Infra Overview" or as complex as making major a new dashboard for parents.

> this is the name of the virtual room that highlights that this about your local agent. the number of nodes it shows will depend if you have children streaming to it but it maps to the parent agent you're accessing in your case

I read the docs on "Organization" and I guess I still got mixed up. I think now I do remember it was the parents name at the top and "local" was the room name. Either way, it would be nice to control the room name when streaming to parents. I know its a cloud feature to be able to create one from the dashboard but I think being able to specify the name when you setup the API key on the parent would be a really nice feature. We could make the default "infra" and it might help with the "local" being somewhat confusing (at least for non-cloud users).

>  you can see that on "systems" you get a summary that allows you to see which systems (nodes) contribute more in terms of volume to the data you're seeing

ahh I did not click on that sub menu or if I did, I was looking for something else and didn't pay any attention to what it does. This will be useful.

> What do you mean? You wanted to have a single parent with child nodes from prod and non-prod?
In this case, it seems it would be more proper to have dedicated parents for each environment, no?

Dedicated parents is our plan but I wouldn't be surprised if someone wanted to run multiple environments from a single parent and have  seperate "Rooms" for each group of infra. For example we have 3 qa environments. It might ve handy to have a dedicated parent for all 3 Qa nodes and give that team a single plane of glass to see their infra but still have it separated by the environment. I'm guessing that is a cloud feature, it's probably even a free one. But so far my team has been somewhat hesitant to use the Netdata Cloud feature even after re-assuring them the data stays on-prem.

**Comment by hugovalente-pm** [comment]
> Yeah I figured it out, but I know I'm not the only one that has complained. My suggestion is to make the dashboards slightly different so thats its easier to distigush a single node dashboard from multiple nodes. It could be as simple as changing "System Overview" to "Infra Overview" or as complex as making major a new dashboard for parents.

Sure a change of title probably could help make it more clear but with the fact that the Single Node view, FR for this - https://github.com/netdata/netdata-cloud/issues/898

> I read the docs on "Organization" and I guess I still got mixed up. I think now I do remember it was the parents name at the top and "local" was the room name. Either way, it would be nice to control the room name when streaming to parents. I know its a cloud feature to be able to create one from the dashboard 

We need to review our [Organize Your Infrastructure and Invite your Team](https://learn.netdata.cloud/docs/monitor-your-infrastructure/organize-your-infrastructure-and-invite-your-team) documentation to better clarify about these _virtual_ spaces and rooms, thanks for the mention

About the customization/organization flexibility that is indeed a thing you already have on Cloud, at the moment there aren't plans to bring such flexibility to the Agent.

> I think being able to specify the name when you setup the API key on the parent would be a really nice feature. We could make the default "infra" and it might help with the "local" being somewhat confusing (at least for non-cloud users).

I'm not sure if there would be implications on making this dynamic since these names are also presented to you If you are on the local agent dashboard and sign-in to Cloud.

> ahh I did not click on that sub menu or if I did, I was looking for something else and didn't pay any attention to what it does. This will be useful.

Nice, I think we have some documentation and videos were we give some tips on how to use the NIDL framework:
* Docs -https://learn.netdata.cloud/docs/visualizations/netdata-charts#nidl-framework
* Introducing Charts v3.0: Slice, dice, filter and pivot the data in any way possible! - https://www.youtube.com/watch?v=T5N_03NV9Ac
* Office Hours episode we talked about this - https://www.youtube.com/watch?v=UNnQMetWDZI

> if someone wanted to run multiple environments from a single parent and have seperate "Rooms" for each group of infra...
> I'm guessing that is a cloud feature, it's probably even a free one. But so far my team has been somewhat hesitant to use the Netdata Cloud feature even after re-assuring them the data stays on-prem.

Yes indeed, this is a Netdata Cloud feature. For teams or companies that are hesitant to use Netdata Cloud we are working on some on-prem options ([comparison here](https://www.netdata.cloud/compare-on-prem/)):
* Fully on-prem - this is more targeted for enterprise level users that have restrictions about using SaaS products, we usually reference this for user with 1,000+ nodes (users will need to setup a similar environment to Netdata Cloud on their premises)
* Netdata Cloud On-Prem Lite - we plan to introduce some space settings that will allow the user to specify, for some dataflows, how they want the data to navigate. Example on charts data: if directly from Agent --> Browser or Agent --> Netdata Cloud --> Browser
